dali/python/nvidia/dali/pipeline.py

```diff
@@ -1528,23 +1603,27 @@ def my_pipe():
 
         pipe = my_pipe(batch_size=42, num_threads=3)
         ...
+
+    Keyword args
+    ------------
+
+    enable_conditionals : bool, optional
+        Enable support for conditional execution of DALI operators using ``if`` statements
+        in the pipeline definition, by default False.
```

Comment:
I understand, that `enable_conditionals` flag has been added to the `pipeline_def`, because it is supported only in decorator mode. Have you considered adding it to the `Pipeline` constructor and handle it there?

The purpose from my perspective would be to accumulate all pipeline parameters in one place (`Pipeline` ctor) and make the decorator pass-through only.

Reply:
The problem is, the using `Pipeline` may direct people to use the stateful pipeline definition via `define_graph`, where one can define helper methods that utilize the `self` as a state to keep some DataNodes. And it is not recommended to use, as it may either error out during tracing or produce a wrong trace.